src/appindexing.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+goog.provide('appindexing');
+
+goog.require('goog.json');// jshint unused:false
+goog.require('utils');
+
+
+appindexing.state = {};
+appindexing.state['androidAppIndexingTagsPresent'] = false;
+appindexing.state['iOSAppIndexingTagsPresent'] = false;
+appindexing.state['androidDetailsComplete'] = false;
+appindexing.state['iOSDetailsComplete'] = false;
+appindexing.state['addediOSTagDom'] = false;
+appindexing.state['addedAndroidTagDom'] = false;
+
+appindexing.checkForAppIndexingTags = function() {
+	var linkTags = document.getElementsByTagName('link');
+	var numLinkTags = linkTags.length;
+
+	for (var counter = 0; counter < numLinkTags; counter++) {
+		var currLinkTag = linkTags[counter];
+		var currLinkTagHref = currLinkTag.href;
```

Comment:
You could potentially throw an error by trying to call .include() on undefined.
`if (!currLinkTagHref) { continue; }`